doc/sphinx/language/extensions/compil-steps.rst

```diff
@@ -0,0 +1,80 @@
+.. _compil-steps:
+
+======================
+Main Compilation Steps
+======================
+
+It is in theory possible to write down every Gallina term explicitly
+as described in the :ref:`Core Language <core-language>` part of this
+manual, in the way they are understood by the kernel, for instance
+:n:`Nat.add (S O) (Nat.mul (S (S O)) (S (S O)))`. However, this would
+be very tedious and error-prone and takes us away from our usual
+mathematical practice. To circumvent this, Rocq offers multiple
+preprocessing mechanisms to help fill the gap between what the users
+would like to input to the system and the fully formal core language
+expected by the kernel.
+
+For instance, the notation mechanisms reflects the eponym mathematical
+practice and allows to write :n:`1 + 2 * 2` instead of the above
+term. Those mechanisms range from simple :ref:`Abbreviations` to full
```

Comment:
```suggestion
For instance, the notation mechanism lets you
write :n:`1 + 2 * 2` instead of the longer and less readable term shown above.
Those mechanisms range from simple :ref:`Abbreviations` to full
```

Eponymous is the adjective form that's grammatically correct, but this should be said in a simpler language.

Reply:
Unfortunately your suggestion is loosing the idea that notation is the notation concept mathematician are familiar with.

Reply:
Then adapt my suggestion.  "eponymous mathematical practice" is a mouthful and vague.  How about "the notation mechanism lets you use forms similar to common mathematical notation"?  (There are differences, e.g. for subscripts and superscripts, summations don't have a large sigma, etc.)

Reply:
Yes, this is again missing the point: what's similar is the concept, not necessarily the exact notations (which are out of the scope here anyway).